sdk/src/wallet/wallet/operations/stronghold.rs

```diff
@@ -3,26 +3,34 @@
 
 use std::time::Duration;
 
-use crate::{client::secret::SecretManager, wallet::Wallet};
+use zeroize::Zeroize;
+
+use crate::{
+    client::secret::{types::Password, SecretManager},
+    wallet::Wallet,
+};
 
 impl Wallet {
     /// Sets the Stronghold password
-    pub async fn set_stronghold_password(&self, password: &str) -> crate::wallet::Result<()> {
+    pub async fn set_stronghold_password(&self, password: impl Into<Password> + Send) -> crate::wallet::Result<()> {
         if let SecretManager::Stronghold(stronghold) = &mut *self.secret_manager.write().await {
-            stronghold.set_password(password).await?;
+            stronghold.set_password(password.into()).await?;
```

Comment:
I think we should be extra careful and do the .into() FIRST thing in the function. What if we don't go in the if?

Reply:
you are right...that's so subtle and not trivially solved by just calling it before the if. If we don't go in the if there's no guarantee that LLVM will execute this line at all, because it will just see that the converted type is not going to be used.

Reply:
If it's before the if, I'm pretty sure we have the guarantee it'll be executed.

Reply:
no, I am afraid we don't as LLVM is allowed to reorder instructions as it pleases in release mode. But there's an easy fix how we can make sure that the given argument is zeroized. Pushing it in a bit.

Reply:
Well anyway, surely it's better at the top of the function

Reply:
I realized that IMHO it's much easier to call `.into()` on the caller side than to take an `impl` as an argument when it comes to zeroizable data like this, because if in some path `.into()` is NOT called, then WE would be responsible for zeroizing the argument which leads to ugly `if ... else`s and mutable arguments.

Reply:
Are you really 100% sure into wouldn't be called? Because I'm not.

Reply:
I am almost certain that compilers nowadays are very good at detecting which lines are redundant, and even if they wouldn't yet there's no guarantee they will in the future. @Alexandcoats What's your take on this?

Reply:
I'll push what I changed and you can take another look, Thibault.

Reply:
Well you already what I'm going to say and I have to say it, but it make UX even worse that it already was 😬

Reply:
can be closed IMO